sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -81,4 +81,8 @@ object TimestampFormatter {
   def apply(timeZone: TimeZone): TimestampFormatter = {
     apply(defaultPattern, timeZone, defaultLocale)
   }
+
+  def getFractionFormatter(timeZone: TimeZone): TimestampFormatter = {
+    apply(DateTimeFormatterHelper.fractionPattern, timeZone, defaultLocale)
```

Comment:
OK, last nit then: this should probably use the naming convention for constants, `FRACTION_PATTERN`.
How about just a method to return this formatter, rather than a special constant that causes another method to return it?

Reply:
Maybe it makes sense to inherit from `Iso8601TimestampFormatter` and override `formatter`. I will try and see how it will look like.

Reply:
I have done this. Please, take a look at the last commit.

Reply:
I have to support only one locale `Locale.US` but I think it is ok for this case. If we will need the fraction formatter for multiple locales, we can come back to the original solution with caching.